tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -174,6 +175,116 @@ DECL_CONVERT_OP(FakeQuant);
 
 #undef DECL_CONVERT_OP
 
+LogicalResult ConvertTFLGeluOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_gelu_op = cast<TFL::GeluOp>(op);
+  Location loc = op->getLoc();
+
+  Value input = tfl_gelu_op.input();
+  ShapedType input_type = input.getType().dyn_cast<ShapedType>();
+  ShapedType output_type =
+      tfl_gelu_op.getResult().getType().dyn_cast<ShapedType>();
+  // Not a ranked tensor output
+  if (!input_type || !output_type) return failure();
+
+  auto input_shape = input_type.getShape();
+  auto output_shape = output_type.getShape();
+
+  bool input_is_qtype =
+      input_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+  bool output_is_qtype =
+      output_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+
+  if (input_is_qtype != output_is_qtype) {
+    return rewriter.notifyMatchFailure(
+        op,
+        "ConvertTFLGeluOp: input/output tensor should be all quantized or all "
+        "floating-point.");
+  }
+
+  if (output_is_qtype) {
+    UniformQuantizedType input_qtype =
+        input_type.getElementType()
+            .dyn_cast<mlir::quant::UniformQuantizedType>();
+    UniformQuantizedType output_qtype =
+        output_type.getElementType()
+            .dyn_cast<mlir::quant::UniformQuantizedType>();
+
+    // The formal definition of gelu.
+    auto gelu_func = [](double x) -> double {
+      return 0.5 * x * (1.0 + std::erf(x / std::sqrt(2)));
+    };
+
+    // Current tfl.gelu only support 8-bit quantized type.
+    if (input_qtype.getStorageTypeIntegralWidth() != 8) return failure();
+
+    Value table_const = getTosaConst8bitTable(
+        rewriter, op, input_qtype.getScale(), input_qtype.getZeroPoint(),
+        output_qtype.getScale(), output_qtype.getZeroPoint(), gelu_func);
+
+    CreateReplaceOpAndInfer<tosa::TableOp>(rewriter, op, output_type, input,
+                                           table_const);
```

Comment:
Return here and then the else clause just becomes the code post the if.

Reply:
Done.